app/models/staged/sponsors.rb

```diff
@@ -19,8 +17,8 @@ def valid?
     def sponsors_valid?
       # Ask them all to validate...
       sponsors.each { |sponsor| sponsor.valid? }
-      # ...return if any have an error
-      sponsors.any? { |sponsor| sponsor.errors.any? }
+      # ...return true if there are no errors at all
+      sponsors.all? { |sponsor| sponsor.errors.none? }
```

Comment:
Oh... that is neater!  Good call.  I think the version we have now is too verbose, I'll change it.